pkg/operator/ceph/pool/radosnamespace/controller.go

```diff
@@ -261,6 +262,45 @@ func (r *ReconcileCephBlockPoolRadosNamespace) reconcile(request reconcile.Reque
 		return reconcile.Result{}, errors.Wrap(err, "failed to save cluster config")
 	}
 
+	poolAndRadosNamespaceName := fmt.Sprintf("%s/%s", pool, getRadosNamespaceName(cephBlockPoolRadosNamespace))
+	if cephBlockPoolRadosNamespace.Spec.Mirroring.Enabled {
```

Comment:
IMO, it might be best to extract these into helper functions so the `reconcile()` function isn't excessively long.

Reply:
Do you mean to move all the mirroring logic to a new function?

Reply:
Agreed, how about a `reconcileMirroring()` method where you move all this new code? It keeps the main `reconcile()` method more maintainable. In an ideal world, methods should approximately fit on a single screen. If bigger, they can commonly be refactored. It really helps with code maintenance when methods are not super long.